### PR TITLE
abstract identity-wallet as generic rpc-connection

### PR DESCRIPTION
**Pull request by pedrouid**

### CODE REVIEW

**Comment by pedrouid** [comment]
Ideally we should try to consolidate the threeIdProvider and replace with this one: https://github.com/pedrouid/3id-provider

**Comment by oed** [comment]
Thanks @pedrouid, this looks great! Definitely the direction to go in. Will circle back to this once I've had more time to think about the `did-provider` spec and how this fits together :)

**Comment by pedrouid** [comment]
Also make sure to checkout https://github.com/pedrouid/3id-provider

File: src/rpc-connection.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+const callbackOrThrow = (callback, errMsg) => {
+  if (callback) {
+    callback(errMsg)
+  } else {
+    throw errMsg instanceof Error ? errMsg : new Error(errMsg)
+  }
+}
+
+class ThreeIdProvider {
```
Comment (oed) [inline src/rpc-connection.js:22]: Class should be renamed to `RpcConnection`.

File: src/identity-wallet.js [hunk 2]
```diff
@@ -42,7 +43,8 @@ class IdentityWallet {
    * @return    {ThreeIdProvider}                   The 3IDProvider for this IdentityWallet instance
    */
   get3idProvider () {
-    return new ThreeIdProvider(this)
+    const connection = new IdWalletRpcConnection(this)
+    return new ThreeIdProvider(connection)
   }
```
Comment (oed) [inline src/identity-wallet.js:48]: Maybe add a method called `getRpcConnection`?

File: src/rpc-connection.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  async open() {
+    // do nothing
+  }
+
+  async close() {
+    // do nothing
+  }
+}
+
+module.exports = ThreeIdProvider
```
Comment (oed) [inline src/rpc-connection.js:114]: This should export RpcConnection.

**Comment by oed** [comment]
This should now be mostly addressed here: https://github.com/3box/identity-wallet-js/pull/56

**Pull request closed without merging**